File: typescript-connector/src/models/AsRealizedProductionFlow.ts
// Class from DFC Business Ontology: #AsRealizedProductionFlow
import { SemanticObject } from "../core/SemanticObject.js";
import { ProductionFlow, type ProductionFlowParams } from "./ProductionFlow.js";

export interface AsRealizedProductionFlowParams extends ProductionFlowParams {}
export class AsRealizedProductionFlow extends ProductionFlow {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:AsRealizedProductionFlow";
  }



  constructor(
    semanticId: string,
    params?: AsRealizedProductionFlowParams,
  ) {
    super(semanticId, params);
    this.semanticType = AsRealizedProductionFlow.SEMANTIC_TYPE;
  }
  static {
    SemanticObject.typeRegistry.set(AsRealizedProductionFlow.SEMANTIC_TYPE, AsRealizedProductionFlow);
  }
}

File: ruby-gem/lib/models/as_realized_production_flow.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #AsRealizedProductionFlow
require_relative '../core/semantic_object'
require_relative 'production_flow'


module DfcLinkmlConnector
  module Models
    class AsRealizedProductionFlow < ProductionFlow
      SEMANTIC_TYPE = "dfc-b:AsRealizedProductionFlow".freeze

    end
  end
end
